src/main/java/org/spongepowered/api/item/inventory/custom/InventoryArchetype.java

```diff
@@ -24,53 +24,58 @@
  */
 package org.spongepowered.api.item.inventory.custom;
 
+import org.spongepowered.api.CatalogType;
 import org.spongepowered.api.Sponge;
-import org.spongepowered.api.item.inventory.type.OrderedInventory;
+import org.spongepowered.api.item.inventory.InventoryProperty;
+import org.spongepowered.api.text.Text;
 import org.spongepowered.api.text.translation.Translation;
-import org.spongepowered.api.util.Identifiable;
 import org.spongepowered.api.util.ResettableBuilder;
+import org.spongepowered.api.util.annotation.CatalogedBy;
 
-/**
- * A CustomInventory is an inventory that has no backing in the world that is
- * created by plugins.
- */
-public interface CustomInventory extends OrderedInventory, Identifiable {
+@CatalogedBy(InventoryArchetypes.class)
+public interface InventoryArchetype extends CatalogType {
```

Comment:
@Mumfrey 

The schematics API will be adding Archetype and it behaves differently than this. I agree, I'll have WorldArchetype extend Archetype in 1.9.4. Schematics API can adjust accordingly.